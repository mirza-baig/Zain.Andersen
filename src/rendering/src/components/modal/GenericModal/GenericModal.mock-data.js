const defaultData = {
    rendering: {
        dataSource: 'sampledatasource',
        componentName: 'GenericModal',
        placeholders: {
          components: [
            {
              "componentName": "PromoGeneric",
              "uid": "8d0b5ef9-af0c-4874-9f8a-26483046665b",
              "dataSource": "{8D0B5EF9-AF0C-4874-9F8A-26483046665B}",
              "fields": {
                "bottomCaptionDescription": {
                  "value": "\n<span id=\"fld_6649519182FE4419BC5C18EA73D49284_0C243F9602584CEFB0027280B78FB028_en_1_c13e08852c664e6fbd123704a2b98820_196365_edit\" sc_parameters=\"prevent-line-break=true\" class=\"scWebEditInput scEnabledChrome\" scfieldtype=\"single-line text\" scdefaulttext=\"[No text in field]\" sc-part-of=\"field\" scwatermark=\"true\" contenteditable=\"true\">Bottom description goes here and spans to two line text over here dummy text goes here</span>\n\n"
                },
                "bottomCaptionHeadline": {
                  "value": "Bottom top copy goes here and spans to two line text over here dummy text goes here"
                },
                "imageRatio": null,
                "componentSpacing": null,
                "sectionId": {
                  "value": ""
                },
                "imgPosition": {
                  "id": "0a58d643-e0c1-4a6e-a9f6-03d731a55121",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Right",
                  "name": "Right",
                  "displayName": "Right",
                  "fields": {
                    "Value": {
                      "value": "right"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "headlineLevel": {
                  "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                  "name": "Heading 2",
                  "displayName": "Heading 2",
                  "fields": {
                    "Value": {
                      "value": "h2"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "headlineText": {
                  "value": "Headline text goes here  with right image 1:1 "
                },
                "eyebrowLevel": null,
                "eyebrowText": {
                  "value": ""
                },
                "body": {
                  "value": "<p><strong>Lorem Ipsum</strong> is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.</p>"
                },
                "cta1Icon": {
                  "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow",
                  "name": "Arrow",
                  "displayName": "Arrow",
                  "fields": {
                    "Value": {
                      "value": "arrow"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "cta1Link": {
                  "value": {
                    "href": "/Demo-HeroTwoColumn",
                    "linktype": "internal",
                    "text": "Call to Action",
                    "querystring": "",
                    "target": "",
                    "id": "{0AF8C545-074B-466E-93CD-B14BF83DB353}"
                  }
                },
                "cta1Modal": null,
                "cta1ModalLinkText": {
                  "value": ""
                },
                "cta1Style": {
                  "id": "49a23327-0397-4cce-a930-e76918d37c42",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary",
                  "name": "Primary",
                  "displayName": "Primary",
                  "fields": {
                    "Value": {
                      "value": "primary"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "primaryImage": {
                  "value": {
                    "src": "https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Picture/photo-1581141849291-1125c7b692b5.jpg?h=900&iar=0&w=1600",
                    "alt": "Img",
                    "width": "1600",
                    "height": "900"
                  }
                },
                "primaryImageMobile": {
                  "value": {}
                },
                "primaryImageMobileFocusArea": null,
                "primaryVideo": null,
                "eventName": {
                  "value": ""
                },
                "eventType": {
                  "value": ""
                },
                "eventZone": {
                  "value": ""
                }
              }
            },
            {
              "componentName": "PromoGeneric",
              "uid": "8d0b5ef9-af0c-4874-9f8a-26483046665b",
              "dataSource": "{8D0B5EF9-AF0C-4874-9F8A-26483046665B}",
              "fields": {
                "bottomCaptionDescription": {
                  "value": "\n<span id=\"fld_6649519182FE4419BC5C18EA73D49284_0C243F9602584CEFB0027280B78FB028_en_1_c13e08852c664e6fbd123704a2b98820_196365_edit\" sc_parameters=\"prevent-line-break=true\" class=\"scWebEditInput scEnabledChrome\" scfieldtype=\"single-line text\" scdefaulttext=\"[No text in field]\" sc-part-of=\"field\" scwatermark=\"true\" contenteditable=\"true\">Bottom description goes here and spans to two line text over here dummy text goes here</span>\n\n"
                },
                "bottomCaptionHeadline": {
                  "value": "Bottom top copy goes here and spans to two line text over here dummy text goes here"
                },
                "imageRatio": null,
                "componentSpacing": null,
                "sectionId": {
                  "value": ""
                },
                "imgPosition": {
                  "id": "0a58d643-e0c1-4a6e-a9f6-03d731a55121",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Right",
                  "name": "Right",
                  "displayName": "Right",
                  "fields": {
                    "Value": {
                      "value": "right"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "headlineLevel": {
                  "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                  "name": "Heading 2",
                  "displayName": "Heading 2",
                  "fields": {
                    "Value": {
                      "value": "h2"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "headlineText": {
                  "value": "Headline text goes here  with right image 1:1 "
                },
                "eyebrowLevel": null,
                "eyebrowText": {
                  "value": ""
                },
                "body": {
                  "value": "<p><strong>Lorem Ipsum</strong> is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.</p>"
                },
                "cta1Icon": {
                  "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow",
                  "name": "Arrow",
                  "displayName": "Arrow",
                  "fields": {
                    "Value": {
                      "value": "arrow"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "cta1Link": {
                  "value": {
                    "href": "/Demo-HeroTwoColumn",
                    "linktype": "internal",
                    "text": "Call to Action",
                    "querystring": "",
                    "target": "",
                    "id": "{0AF8C545-074B-466E-93CD-B14BF83DB353}"
                  }
                },
                "cta1Modal": null,
                "cta1ModalLinkText": {
                  "value": ""
                },
                "cta1Style": {
                  "id": "49a23327-0397-4cce-a930-e76918d37c42",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary",
                  "name": "Primary",
                  "displayName": "Primary",
                  "fields": {
                    "Value": {
                      "value": "primary"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "primaryImage": {
                  "value": {
                    "src": "https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Picture/photo-1581141849291-1125c7b692b5.jpg?h=900&iar=0&w=1600",
                    "alt": "Img",
                    "width": "1600",
                    "height": "900"
                  }
                },
                "primaryImageMobile": {
                  "value": {}
                },
                "primaryImageMobileFocusArea": null,
                "primaryVideo": null,
                "eventName": {
                  "value": ""
                },
                "eventType": {
                  "value": ""
                },
                "eventZone": {
                  "value": ""
                }
              }
            }
          ]
        }
        },
        "fields": {
          "desktopAnimationStyle": {
            "id": "f4d62dac-74b6-4247-9ecd-88c316281470",
            "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Modal-Animation-Style/Slide-in-from-Top",
            "name": "Slide in from Top",
            "displayName": "Slide in from Top",
            "fields": {
              "Value": {
                "value": "slide-in-top"
              }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
          },
          "displayCloseBar": {
            "value": true
          },
          "mobileAnimationStyle": {
            "id": "09d31a9b-b31f-4209-b746-0691fcf3d51c",
            "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Modal-Animation-Style/Slide-in-from-Bottom",
            "name": "Slide in from Bottom",
            "displayName": "Slide in from Bottom",
            "fields": {
              "Value": {
                "value": "slide-in-bottom"
              }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
          },
          "modalSize": {
            "id": "fe6d9820-0dcd-4cfe-8f89-c2333d243470",
            "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Generic-Modal-Size/Extra-Large",
            "name": "Extra Large",
            "displayName": "Extra Large",
            "fields": {
              "Value": {
                "value": "extra-large"
              }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
          },
          "componentSpacing": null,
          "sectionId": {
            "value": ""
          },
          "modalId": {
            "value": "genericModal3"
          },
          "eventName": {
            "value": ""
          },
          "eventType": {
            "value": ""
          },
          "eventZone": {
            "value": ""
          },
          "children": []
        },
        "placeholders": {
          "components": [
            {
              "uid": "c93358fe-77f9-4af8-9e10-eeecaf8cdae8",
              "componentName": "PromoGeneric",
              "dataSource": "{C93358FE-77F9-4AF8-9E10-EEECAF8CDAE8}",
              "fields": {
                "bottomCaptionDescription": {
                  "value": "<p>Bottom description goes here and spans to two line text over here dummy text goes here</p>"
                },
                "bottomCaptionHeadline": {
                  "value": "Bottom top copy goes here and spans to two line text over here dummy text goes here"
                },
                "imageRatio": null,
                "componentSpacing": null,
                "sectionId": {
                  "value": ""
                },
                "imgPosition": {
                  "id": "0a58d643-e0c1-4a6e-a9f6-03d731a55121",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Right",
                  "name": "Right",
                  "displayName": "Right",
                  "fields": {
                    "Value": {
                      "value": "right"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "headlineLevel": {
                  "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                  "name": "Heading 2",
                  "displayName": "Heading 2",
                  "fields": {
                    "Value": {
                      "value": "h2"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "headlineText": {
                  "value": "Headline goes here with right image 3:2 "
                },
                "eyebrowLevel": null,
                "eyebrowText": {
                  "value": ""
                },
                "body": {
                  "value": "<strong>Lorem Ipsum</strong> is simply dummy text of the printing and\ntypesetting industry.&nbsp;"
                },
                "cta1Link": {
                  "value": {
                    "href": "/Demo-HeroTwoColumn",
                    "linktype": "internal",
                    "text": "Call to Action",
                    "querystring": "",
                    "target": "",
                    "id": "{0AF8C545-074B-466E-93CD-B14BF83DB353}"
                  }
                },
                "cta1ModalLinkText": {
                  "value": ""
                },
                "cta1Icon": {
                  "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow",
                  "name": "Arrow",
                  "displayName": "Arrow",
                  "fields": {
                    "Value": {
                      "value": "arrow"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "cta1Modal": null,
                "cta1Style": {
                  "id": "49a23327-0397-4cce-a930-e76918d37c42",
                  "url": "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary",
                  "name": "Primary",
                  "displayName": "Primary",
                  "fields": {
                    "Value": {
                      "value": "primary"
                    }
                  },
                  "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                  "templateName": "Enum"
                },
                "cta1AriaLabel": {
                  "value": ""
                },
                "primaryImage": {
                  "value": {
                    "src": "https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Picture/photo-1581141849291-1125c7b692b5.jpg?h=900&iar=0&w=1600",
                    "alt": "Img",
                    "width": "1600",
                    "height": "900"
                  }
                },
                "primaryImageMobile": {
                  "value": {}
                },
                "primaryImageMobileFocusArea": null,
                "primaryVideo": null,
                "eventName": {
                  "value": ""
                },
                "eventType": {
                  "value": ""
                },
                "eventZone": {
                  "value": ""
                }
              }
            }
          ]
        }
};

export default defaultData;
