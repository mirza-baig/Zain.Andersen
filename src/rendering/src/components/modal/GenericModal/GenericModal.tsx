// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { Placeholder, RouteData, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { ComponentProps } from 'lib/types/component-props';
import ModalWrapper, { AnimationStyle, ModalSize } from 'src/helpers/ModalWrapper/ModalWrapper';
import { useEffect, useState } from 'react';
import { useModalIdContext } from 'lib/context/GenericModalIDContext';
import { getEnum } from 'lib/utils';

export type GenericModalProps =
  Feature.EnterpriseWeb.Enterprise.Components.Modal.GenericModal.GenericModal & {
    fields: {
      children: RouteData[];
    };
  } & ComponentProps;
const GenericModal = (props: GenericModalProps) => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const { selectedModalId, setSelectedModalId, prevFocusedElementRef } = useModalIdContext();

  const desktopAnimationStyle =
    getEnum<AnimationStyle>(props.fields?.desktopAnimationStyle) || null;
  const mobileAnimationStyle = getEnum<AnimationStyle>(props.fields?.mobileAnimationStyle) || null;

  const displayCloseBar = props.fields?.displayCloseBar?.value || false;

  const handleOpenModal = () => {
    setIsModalOpen(true);
  };

  const handleCloseModal = () => {
    setIsModalOpen(false);
    setSelectedModalId('');
    prevFocusedElementRef?.current?.focus();
  };

  const modalSize = getEnum<ModalSize>(props.fields?.modalSize) ?? 'large';

  useEffect(() => {
    if (selectedModalId) {
      selectedModalId === props.fields?.modalId.value && handleOpenModal();
    }
    if (selectedModalId === '') {
      handleCloseModal();
    }
    // "props.fields?.modalId.value" is coming directly from layout service.
    // "handleCloseModal" is a function which is not going to be changed.
    // We can ignore both suggested deps.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [selectedModalId]);

  return (
    <div data-component="modal/genericmodal">
      <ModalWrapper
        size={modalSize}
        handleClose={handleCloseModal}
        modalLabel={props.fields?.modalId.value}
        isModalOpen={isModalOpen}
        desktopAnimationStyle={desktopAnimationStyle}
        mobileAnimationStyle={mobileAnimationStyle}
        displayCloseBar={displayCloseBar}
      >
        <Placeholder
          isGenericModal={isModalOpen}
          name="components"
          rendering={props.rendering}
          render={(components) => {
            return components.map((component, index) => <div key={index}>{component}</div>);
          }}
        />
      </ModalWrapper>
    </div>
  );
};

export default withDatasourceCheck()<GenericModalProps>(GenericModal);
