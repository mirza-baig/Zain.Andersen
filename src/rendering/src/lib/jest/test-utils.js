// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
