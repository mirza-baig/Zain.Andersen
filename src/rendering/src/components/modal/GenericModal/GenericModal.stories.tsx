// Global
import { Story, Meta } from '@storybook/react';
// Local
import GenericModal, { GenericModalProps } from './GenericModal';
import defaultData from './GenericModal.mock-data';

export default {
  title: 'Modal/GenericModal',
  component: GenericModal,
} as Meta;
const Template: Story<GenericModalProps> = (props) => <GenericModal {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
