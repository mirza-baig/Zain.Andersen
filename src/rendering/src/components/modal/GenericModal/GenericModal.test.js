// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import GenericModal from './GenericModal';
import defaultData from './GenericModal.mock-data';

it('renders correctly', async () => {
  const component = snapshot(GenericModal, { componentProps: defaultData });
  await hasDataComponent(component, 'modal/genericmodal');
});
