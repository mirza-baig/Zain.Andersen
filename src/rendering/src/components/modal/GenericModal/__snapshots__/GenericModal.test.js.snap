// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-component="modal/genericmodal"
>
  <div
    aria-label="genericModal3"
    aria-modal="true"
    class="fixed top-0 left-0 z-[1000] h-screen w-screen transform items-center justify-center bg-black bg-opacity-75 transition-transform duration-500 ease-in-out md:z-[998] flex"
    id="genericModal3"
    role="dialog"
    style="height: calc(100vh - 0px); transform: translateY(calc(-120%));"
    tabindex="-1"
  >
    <div
      class="relative flex flex-col overflow-y-auto bg-white md:mx-auto w-screen h-screen"
      style="height: calc(100vh - 0px);"
      tabindex="0"
    >
      <div
        class="flex h-[40px] w-full justify-end"
      >
        <button
          class="mt-m mr-m"
        >
          <span />
        </button>
      </div>
      <div
        class="!pb-l md:pb-l pb-l"
      >
        <div>
          <div
            style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
          >
            <h2>
              PromoGeneric
            </h2>
            <p>
              JSS component is missing React implementation. See the developer console for more information.
            </p>
          </div>
        </div>
        <div>
          <div
            style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
          >
            <h2>
              PromoGeneric
            </h2>
            <p>
              JSS component is missing React implementation. See the developer console for more information.
            </p>
          </div>
        </div>
      </div>
      <button
        class="sticky bottom-0 z-10 mt-auto flex w-full items-center justify-center gap-xxxs border-t border-gray bg-white py-2.5 md:bg-primary"
      >
        <span
          class="font-serif text-button font-bold"
        >
          Close
        </span>
        <span />
      </button>
    </div>
  </div>
</div>
`;
